src/Integration/Telemetry/TelemetryData.cs

```diff
@@ -19,15 +19,44 @@
  */
 
 using System;
+using System.Xml.Serialization;
 
 namespace SonarLint.VisualStudio.Integration
 {
     public sealed class TelemetryData
     {
         public bool IsAnonymousDataShared { get; set; }
-        public DateTime InstallationDate { get; set; }
-        public DateTime LastSavedAnalysisDate { get; set; }
+
         public long NumberOfDaysOfUse { get; set; }
-        public DateTime LastUploadDate { get; set; }
+
+        [XmlIgnore]
+        public DateTimeOffset InstallationDate { get; set; }
+
+        [XmlIgnore]
+        public DateTimeOffset LastSavedAnalysisDate { get; set; }
+
+        [XmlIgnore]
+        public DateTimeOffset LastUploadDate { get; set; }
+
+        [XmlElement(nameof(InstallationDate))]
+        public string InstallationDateString
+        {
+            get { return InstallationDate.ToString("o"); }
+            set { InstallationDate = DateTimeOffset.ParseExact(value, "o", null); }
+        }
+
+        [XmlElement(nameof(LastSavedAnalysisDate))]
+        public string LastSavedAnalysisDateString
+        {
+            get { return LastSavedAnalysisDate.ToString("o"); }
+            set { LastSavedAnalysisDate = DateTimeOffset.ParseExact(value, "o", null); }
+        }
+
+        [XmlElement(nameof(LastUploadDate))]
+        public string LastUploadDateString
+        {
+            get { return LastUploadDate.ToString("o"); }
+            set { LastUploadDate = DateTimeOffset.ParseExact(value, "o", null); }
```

Comment:
Looks like something that can throw if the format of the string is not good. I think it should be handled.